scripts/listapranzo.js

```diff
@@ -85,6 +85,51 @@ module.exports = function (robot) {
     order.users[user.id] = dishes;
   };
 
+  var countMatches = function (str1, str2) {
+    var s1 = str1.split(' ');
+    var s2 = str2.split(' ');
+
+    s1 = s1.map(function (s) { return s.trim().toLowerCase(); });
+    s2 = s2.map(function (s) { return s.trim().toLowerCase(); });
+
+    var match = 0;
+
+    for (i = 0; i < s1.length; i++) {
+      for (j = 0; j < s2.length; j++) {
+        if (s1[i] === s2[j])
```

Comment:
Non sono sicuro di aver capito cosa intendi... il sistema di matching mi sembra un po' troppo aperto, ma soprattutto vorrei poter continuare a usare un solo comando per fare l'ordine... se lui mi risponde sì/no, poi cosa succede? 

Un'alternativa più semplice che posso proporre è quella di non aggiungere l'ordine in automatico a meno che non vengono trovate tutte le parole inserite. Per cui tornando al tuo esempio una ricerca per "pasta pesto" quando c'è "pasta al pomodoro" ritornerebbe una richiesta di chiarimenti.

L'assunto è che gli utenti chiedano qualcosa di saliente del piatto ("per me pesto", quando nel  menù c'era "garganelli di portofino al pesto"), per cui credo possa essere sufficiente, che ne dici?